students/john_rogers/lesson09/mailroom.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+"""
+mailroom.py: use classes where applicable
+Author: JohnR
+Version: 1.0
+Last updated: 2/17/19
+Notes:
+"""
+
+from donors import DonorDataBase as DDB
+from donors import Donor as D
+
+
+def main():
+    """
+    Main script loop and user interaction
+    Start with a database of several donors and donations
+    :return: N/A
+    """
+
+    donor_db = DDB()
+
+    d1 = D('John', 'Randal', [12.32, 34.53, 532.32])
+    d2 = D('Sarah', 'Samson', [1.32, 324.53, 2345.33, 6602.12])
+    d3 = D('Alex', 'Rez', [122.32, 2334.53])
+    d4 = D('Billy', 'Durst', [15.32, 34.00])
+
+    donor_db.add_donor(d1)
+    donor_db.add_donor(d2)
+    donor_db.add_donor(d3)
+    donor_db.add_donor(d4)
+
+    main_prompt = (
+        "\nWelcome to the main menu!\n"
+        "Please pick a number from the following:\n"
+        "1: exit the program\n"
+        "2: check donor list and become a donor\n"
+        "3: display a summary of current donor activity\n"
+        "4: print out a thank you for each donor\n"
+        "5: save a thank you note to disk for each donor\n"
+        ">>> "
+    )
+
+    main_dispatch = {
+        '1': exit_menu,
+        '2': donor_actions,
+        '3': DDB.print_summary,
+        '4': DDB.thank_all,
+        '5': DDB.save_report,
+    }
+
+    menu(main_prompt, main_dispatch, donor_db)
+
+
+def menu(main_prompt, main_dispatch, donor_db):
+    """
+    Get user input
+    :return: call the appropriate menu item
+    """
+    while True:
+        response = input(main_prompt)
+        try:
+            main_dispatch[response](donor_db)
+        except KeyError:
+            print('Please enter a number between 1 and 5.')
+
+
+def exit_menu(data):
+    print()
+    save = input('Enter Y to save all data to disk before exiting. ')
+    save = save.lower()
+    if save == 'y':
+        DDB.save_report(data)
```

Comment:
`DonorDatabase.save_report` is a function (object)
`DonorDatabase().save_report` is a method of the class

since you're using OO you should be working with an instance and executing a method of the instance : `donor_db.save_report()`

you can use `DonorDatabase.save_report()` with staticmethods which don't work on the object/instance and are simple functions that relate to the object and that's why one would define them inside of a class.